### ISSUE TITLE
Lua 5.3 support

### ISSUE THREAD

**Issue by Fale** [leafo/pgmoon#54]
I'm trying to run pgmoon on a Lua 5.3 installation, and I receive the following:

```
lua: /usr/share/lua/5.3/pgmoon/init.lua:6: module 'bit' not found:
	no field package.preload['bit']
	no file '/usr/share/lua/5.3/bit.lua'
	no file '/usr/share/lua/5.3/bit/init.lua'
	no file '/usr/lib64/lua/5.3/bit.lua'
	no file '/usr/lib64/lua/5.3/bit/init.lua'
	no file './bit.lua'
	no file './bit/init.lua'
	no file '/usr/lib64/lua/5.3/bit.so'
	no file '/usr/lib64/lua/5.3/loadall.so'
	no file './bit.so'
stack traceback:
	[C]: in function 'require'
	/usr/share/lua/5.3/pgmoon/init.lua:6: in main chunk
	[C]: in function 'require'
	pgmoon.lua:1: in main chunk
	[C]: in ?
```

I think the code should be tweaked to accept the external bit library (for Lua 5.1 and Lua 5.2) and the internal one for Lua 5.3

**Comment by Fale** [comment]
This seems to be related: https://github.com/luarocks/luarocks/issues/526

**Comment by leafo** [comment]
btw, when we do fix this, we need to turn on lua5.3 on ci:

https://github.com/leafo/pgmoon/blob/master/.travis.yml#L14

If you enable it and run ci now you can see why it's currently not working

**Comment by Fale** [comment]
I'm pretty new to Lua, but from my understanding, [this line](https://github.com/leafo/pgmoon/blob/master/pgmoon/init.lua#L6) should not contain the word 'require'.
Also, the following line probably needs some small changes since - from what I read - the integrated "bit" library is not drop in.
I tried to propose a patch for this today, but I did not found anywhere an example on how to identify the lua version at runtime. I think the clean way to support 5.1, 5.2, and 5.3 would be to identify the version and behave differently based on that

**Comment by leafo** [comment]
Sorry, I left out some information. There are other reasons why specs don't pass on 5.3, it's not a simple change to `require`. If you want to get around the require error you can install `luabitop` from luarocks for your corresponding version of lua.

It's listed as a dependency: https://github.com/leafo/pgmoon/blob/master/pgmoon-dev-1.rockspec#L16

I'm guessing you didn't install pgmoon via luarocks?

**Comment by ttfkam** [comment]
Should pgmoon migrate from luabitop to bit32 now that Lua 5.2+ supports native 32 integers and related ops?

**Comment by RyanSquared** [comment]
https://github.com/AlberTajuelo/bitop-lua looks like a neat library that includes a luabitop compatibility layer

**Comment by tholu** [comment]
I would support a fix for this as well, this blocks Lua 5.3 compatibility e.g. for https://github.com/Kong/kong.

**Comment by RyanSquared** [comment]
> If you want to get around the require error you can install luabitop from luarocks for your corresponding version of lua.

I now remember why I wrote a wrapper library for this purpose - you actually *can't* do that. The dependencies of luabitop requires lua >= 5.1, lua < 5.3.

I wrote [this](https://github.com/RyanSquared/pgmoon/commit/93eb0ae16f39da5a1ea82cca14ce0542668b6d44) a while ago for minimal support. I can rebase it from the latest master and people can specify that in their rockspecs until then.
